src/components/Article.jsx

```diff
@@ -0,0 +1,95 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import formatDate from "../assets/utils";
+
+function Article({ users }) {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+  const [comments, setComments] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(true);
+
+    const fetchArticle = axios.get(
+      `https://news-api-project-hj1l.onrender.com/api/articles/${article_id}`
+    );
+
+    const fetchComments = axios.get(
+      `https://news-api-project-hj1l.onrender.com/api/articles/${article_id}/comments`
+    );
+
+    Promise.all([fetchArticle, fetchComments]).then((response) => {
+      const { article } = response[0].data;
+      const { comments } = response[1].data;
+      setArticle(article);
+      setComments(comments);
+      setLoading(false);
+    });
+  }, [article_id]);
+
+  if (loading) {
+    return (
+      <div>
+        <h3>Loading ...</h3>;
+      </div>
+    );
+  }
+
+  const userIndex = users.findIndex((user) => {
+    return user.username === article.author;
+  });
+  const user = users[userIndex];
+
+  return (
+    <>
+      <article>
+        <p className="topic">{article.topic}</p>
+        <h3 className="article-title-individual">{article.title}</h3>
+        <div className="article-details-individual">
+          <img
+            src={user.avatar_url}
+            alt="User avatar"
+            className="article-user-icon"
+          />
+          <div className="author-date-individual">
+            <p>{user.name}</p>
+            <p>{formatDate(article.created_at)}</p>
+          </div>
+          <p>Votes {article.votes}</p>
+        </div>
+        <img src={article.article_img_url} alt={article.title} />
+        <p>{article.body}</p>
+      </article>
+      <section>
+        <h4 className="comments-header">Comments</h4>
+        {comments.map((comment, index) => {
+          const userIndex = users.findIndex((user) => {
+            return user.username === comment.author;
```

Comment:
not sure why you need to do this, you're checking if the author is in the users array but they have to be in order to post a comment so this is the same thing as just displaying the comment.author?